pydantic/types.py

```diff
@@ -474,7 +474,7 @@ class ConstrainedExample(BaseModel):
         '''
     ```
     """  # noqa: D212
-    return Annotated[
+    return Annotated[  # pyright: ignore[reportReturnType]
```

Comment:
Same question - why is this needed here?

Reply:
Maybe @adriangb would know - you're up to date on typing with `Annotated` 🌟

Reply:
This seems fine actually, these functions are already so hacky anyways, I'm not surprised this is a problem.